lib/awesome_nested_set/tree.rb

```diff
@@ -23,6 +23,7 @@ def rebuild!
             # setup index for this scope
             indices[scope_for_rebuild.call(root_node)] ||= 0
             set_left_and_rights(root_node)
+            reset_counter_cache(root_node)
```

Comment:
This will only run if the model isn't valid. There are a couple options here:

1. Add a conditional validation for the counter cache column
2. Run this outside of the validation (before we check `model.valid?`)
3. Create a separate method entirely outside of `rebuild!` for this functionality